nwbwidgets/image.py

```diff
@@ -1,94 +1,173 @@
+import numpy as np
 import plotly.graph_objects as go
 import pynwb
-from ipywidgets import widgets, fixed, Layout
+from ipywidgets import widgets, Layout
 from pynwb.image import GrayscaleImage, ImageSeries, RGBImage
-from tifffile import imread, TiffFile
 
 from .base import fig2widget
-from .controllers import StartAndDurationController
+from .utils.cmaps import linear_transfer_function
+from .utils.imageseries import get_frame_count, get_frame
 from .utils.timeseries import (
     get_timeseries_maxt,
     get_timeseries_mint,
     timeseries_time_to_ind,
 )
 
+PathType = Union[str, Path]
+
 
 class ImageSeriesWidget(widgets.VBox):
     """Widget showing ImageSeries."""
 
     def __init__(
         self,
         imageseries: ImageSeries,
-        foreign_time_window_controller: StartAndDurationController = None,
-        **kwargs
+        foreign_time_slider: widgets.FloatSlider = None,
+        neurodata_vis_spec: dict = None,
     ):
         super().__init__()
         self.imageseries = imageseries
-        self.controls = {}
-        self.out_fig = None
+        self.figure = None
+        self.time_slider = foreign_time_slider
 
-        # Set controller
-        if foreign_time_window_controller is None:
-            tmin = get_timeseries_mint(imageseries)
-            if imageseries.external_file and imageseries.rate:
-                tif = TiffFile(imageseries.external_file[0])
-                tmax = imageseries.starting_time + len(tif.pages) / imageseries.rate
-            else:
-                tmax = get_timeseries_maxt(imageseries)
-            self.time_window_controller = StartAndDurationController(tmax, tmin)
+        if imageseries.external_file is not None:
+
+            # set time slider:
+            tmax = (
+                imageseries.starting_time
+                + get_frame_count(imageseries.external_file[0]) / imageseries.rate
+            )
+            if self.time_slider is None:
+                self.time_slider = widgets.FloatSlider(
+                    min=imageseries.starting_time,
+                    max=tmax,
+                    orientation="horizontal",
+                    description="time(s)",
+                )
+            external_file = imageseries.external_file[0]
+            self.file_selector = None
+            # set file selector:
+            if len(imageseries.external_file) > 1:
+                self.file_selector = widgets.Dropdown(options=imageseries.external_file)
+                external_file = self.file_selector.value
+
+                def update_time_slider(value):
+                    path_ext_file = value["new"]
+                    # Read first frame
+                    nonlocal external_file
+                    external_file = path_ext_file
+                    tmax = (
+                        imageseries.starting_time
+                        + get_frame_count(path_ext_file) / imageseries.rate
+                    )
+                    tmin = 0
```

Comment:
why reset tmin to 0 every update?

Reply:
knowing the `starting_time` for video files other than the first one is not possible. Unless we assume the starting time to the end time for the last video in the sequence. But I don't think this would be a good approximation, so I have made  the `time_slider` widget to span the video frames (0, num_frames) instead, only when the user requests video other than the first one.